packaging/linux/buildpackage.sh

```diff
@@ -110,63 +104,70 @@ else
 	wget -cq https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage || exit 3
 fi
 
-tar xf "${PACKAGING_APPIMAGE_DEPENDENCIES_TEMP_ARCHIVE_NAME}"
-chmod a+x appimagetool-x86_64.AppImage
-
 echo "Building AppImage"
 
-install -d "${BUILTDIR}/usr/bin"
-install -d "${BUILTDIR}/etc/mono/4.5"
-install -d "${BUILTDIR}/usr/lib/mono/4.5"
+tar xf "${PACKAGING_APPIMAGE_DEPENDENCIES_TEMP_ARCHIVE_NAME}" -C "${APPDIR}"
+chmod 0755 "${APPDIR}/usr/bin/mono"
+chmod 0644 "${APPDIR}/etc/mono/config"
+chmod 0644 "${APPDIR}/etc/mono/4.5/machine.config"
+chmod 0644 "${APPDIR}/usr/lib/mono/4.5/Facades/"*.dll
+chmod 0644 "${APPDIR}/usr/lib/mono/4.5/"*.dll "${APPDIR}/usr/lib/mono/4.5/"*.exe
+chmod 0755 "${APPDIR}/usr/lib/"*.so
 
-install -Dm 0755 usr/bin/mono "${BUILTDIR}/usr/bin/"
-
-install -Dm 0644 /etc/mono/config "${BUILTDIR}/etc/mono/"
-install -Dm 0644 /etc/mono/4.5/machine.config "${BUILTDIR}/etc/mono/4.5"
-
-for f in $(ls usr/lib/mono/4.5/*.dll usr/lib/mono/4.5/*.exe); do install -Dm 0644 "$f" "${BUILTDIR}/usr/lib/mono/4.5/"; done
-for f in $(ls usr/lib/*.so usr/lib/*.so.*); do install -Dm 0755 "$f" "${BUILTDIR}/usr/lib/"; done
-
-rm -rf libs libs.tar.bz2
+rm -rf "${PACKAGING_APPIMAGE_DEPENDENCIES_SOURCE}"
 
 # Add launcher and icons
-sed "s/{MODID}/${MOD_ID}/g" include/AppRun.in | sed "s/{MODNAME}/${PACKAGING_DISPLAY_NAME}/g" > AppRun.temp
-install -m 0755 AppRun.temp "${BUILTDIR}/AppRun"
+sed "s/{MODID}/${MOD_ID}/g" "${TEMPLATE_ROOT}/${ENGINE_DIRECTORY}/packaging/linux/AppRun.in" | sed "s/{MODNAME}/${PACKAGING_DISPLAY_NAME}/g" > "${APPDIR}/AppRun"
+chmod 0755 "${APPDIR}/AppRun"
+
+if [ -n "${PACKAGING_DISCORD_APPID}" ]; then
+	sed "s/{DISCORDAPPID}/${PACKAGING_DISCORD_APPID}/g" "${TEMPLATE_ROOT}/${ENGINE_DIRECTORY}/packaging/linux/openra.desktop.discord.in" > temp.desktop.in
+	sed "s/{DISCORDAPPID}/${PACKAGING_DISCORD_APPID}/g" "${TEMPLATE_ROOT}/${ENGINE_DIRECTORY}/packaging/linux/openra-mimeinfo.xml.discord.in" > temp.xml.in
+else
+	cp "${TEMPLATE_ROOT}/${ENGINE_DIRECTORY}/packaging/linux/openra.desktop.in" temp.desktop.in
+	cp "${TEMPLATE_ROOT}/${ENGINE_DIRECTORY}/packaging/linux/openra-mimeinfo.xml.in" temp.xml.in
+fi
 
-sed "s/{MODID}/${MOD_ID}/g" include/mod.desktop.in | sed "s/{MODNAME}/${PACKAGING_DISPLAY_NAME}/g" | sed "s/{TAG}/${TAG}/g" > temp.desktop
-install -Dm 0755 temp.desktop "${BUILTDIR}/usr/share/applications/openra-${MOD_ID}.desktop"
-install -m 0755 temp.desktop "${BUILTDIR}/openra-${MOD_ID}.desktop"
+mkdir -p "${APPDIR}/usr/share/applications"
+chmod 0755 temp.desktop.in
```

Comment:
These are text files. They don't need executable file permissions.

Reply:
https://wiki.ubuntu.com/SecurityTeam/Policies#Execute-Permission_Bit_Required says that .desktop files do need the execute bit?

Reply:
Must be an Ubuntu specific thing. On @openSUSE none have this and I am not certain if it even causes a `spurious-executable` rpmlint warning.